File: Professional Image Modifier 4/[VB]-Professional Image Modifier 4/VB/ImageFunctions/Modifications/CornerDetection/Harris.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Accord.Imaging
Imports AForge

Namespace ImageFunctions.Modifications.CornerDetection
	Friend Class Harris

		Private CurrentImage As String

		Public Delegate Sub ImageCompleteHandler(ByVal Corners As List(Of IntPoint))
		Public Event ImageComplete As ImageCompleteHandler

		Public Sub New(ByVal CurrentImage As String)
			Me.CurrentImage = CurrentImage
		End Sub

		Public Sub GetCorners(ByVal threshold As Integer, ByVal sigma As Integer)
			' create corners detector's instance
			Dim hcd As New HarrisCornersDetector(HarrisCornerMeasure.Noble,threshold,sigma)

			' Apply the filter and return the points
			Dim corners As List(Of IntPoint) = hcd.ProcessImage(AForge.Imaging.Image.FromFile(Me.CurrentImage))
			RaiseEvent ImageComplete(corners)
		End Sub

	End Class
End Namespace

File: Professional Image Modifier 4/[C#]-Professional Image Modifier 4/C#/ImageFunctions/Modifications/CornerDetection/Harris.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Accord.Imaging;
using AForge;

namespace ImageFunctions.Modifications.CornerDetection
{
	class Harris
	{

		private string CurrentImage;

		public delegate void ImageCompleteHandler(List<IntPoint> Corners);
		public event ImageCompleteHandler ImageComplete;

		public Harris(string CurrentImage)
		{
			this.CurrentImage = CurrentImage;
		}

		public void GetCorners(int threshold, int sigma)
		{
			// create corners detector's instance
			HarrisCornersDetector hcd = new HarrisCornersDetector(HarrisCornerMeasure.Noble,threshold,sigma);
				
			// Apply the filter and return the points
			List<IntPoint> corners = hcd.ProcessImage(AForge.Imaging.Image.FromFile(this.CurrentImage));
			if (ImageComplete != null) ImageComplete(corners);
		}

	}
}
